proposals/0002-root-signature-in-clang.md

```diff
@@ -400,7 +399,7 @@ signatures with functions.
 Example for same root signature as above:
 
 ```llvm
-!dx.rootsignatures = !{!2} ; list of function/root signature pairs
+!hlsl.rootsignatures = !{!2} ; list of function/root signature pairs
```

Comment:
Why this change to the design?

Reply:
We have 2 layers of metadata being generated, when the metadata comes from hlsl and goes into llvm ir, we called it an hlsl metadata, so that is why I changed to hlsl. I am following the pattern done for MS Style If attributes, that also generate metadata

Reply:
This is different - this is DX-specific and is similar to things like `!dx.valver` and `!dx.entryPoints`. I don't think we should use a different naming convention for this.